### PR TITLE
Add Job.state to API

### PR DESCRIPTION
**Pull request by abreckner**
**Intent (What)**
Add the ability to update the Job's state via the API.

There is a discrepancy in that this only returns a status and no Job.

### CODE REVIEW

File: lib/xpm_ruby/connection.rb [hunk 1]
```diff
@@ -62,7 +62,7 @@ def handle_response(response)
         when "OK"
           xml["Response"]
         when "ERROR"
-          raise Error.new(response["ErrorDescription"])
+          raise Error.new(xml["Response"]["ErrorDescription"])
```
Comment (abreckner) [inline lib/xpm_ruby/connection.rb:65]: @Subskii just wanted to double check that this change makes sense. The response was a Faraday response which had no ErrorDescription on it, but the xml["Response"] does have one (and at this stage we are guaranteed to have an xml["Response"]).

Comment (Subskii) [reply lib/xpm_ruby/connection.rb:65]: Great catch Tony.

Looks right according to their [docs](https://developer.xero.com/documentation/practice-manager/overview-practice-manager-api). That also explains why I haven't been seeing the error message bubble up in some ailing specs too!

Was surprised the [Staff.list tests](https://github.com/ignitionapp/xpm_ruby/blob/master/spec/xpm_ruby/staff_spec.rb#L29) didn't pick up anything at first but makes sense as we don't assert on the message just class.

Comment (abreckner) [reply lib/xpm_ruby/connection.rb:65]: Ah ok cool thanks!

Comment (abreckner) [reply lib/xpm_ruby/connection.rb:65]: I think we are at the stage now where we should be testing more than just the happy path.
;)

**Pull request merged**